### ISSUE TITLE
CUDA ERROR  while training NER + RELATIONAL model. Found Bug in training/loop.py 

### ISSUE THREAD

**Issue by karndeepsingh** [explosion/spaCy#9753]
I have been training NER + RELATIONAL model on 5K dataset and validating on 2K dataset on 12GB GPU and while training CUDA ERROR was raised and I was debugging in the code to find an error. I found that in training/loop.py file it wasn't calculating any score inside the if condition mentioned below and because of this I feel gradients keeps on accumulating and it doesn't go for another batch for training and thus raises and CUDA ERROR. Please look into this.
```
#print("Step-4")
        optimizer.step_schedules()
        if not (step % eval_frequency):
            print("Step-5")
            if optimizer.averages:
                with nlp.use_params(optimizer.averages):
                    score, other_scores = evaluate()
                    print(score)
            else:
                score, other_scores = evaluate()
            results.append((score, step))
            is_best_checkpoint = score == max(results)[0]
            print("Step-6")
        else:
            score, other_scores = (None, None)
            is_best_checkpoint = None
        words_seen += sum(len(eg) for eg in batch)
        info = {
            "epoch": epoch,
            "step": step,
            "score": score,
            "other_scores": other_scores,
            "losses": losses,
            "checkpoints": results,
            "seconds": int(timer() - start_time),
            "words": words_seen,
        }
```

**Comment by svlandeg** [comment]
Duplicate of https://github.com/explosion/spaCy/discussions/9723 - closing. Please refrain from opening duplicate issues about topics, as this puts an unnecessary additional burden on our team. Thanks for understanding!
